tsdb/wal/wal_test.go

```diff
@@ -484,3 +484,32 @@ func BenchmarkWAL_Log(b *testing.B) {
 		})
 	}
 }
+
+// TestClose ensures that calling Close more than once doesn't panic and doesn't block.
+func TestCloseGoroutineLeak(t *testing.T) {
```

Comment:
Is this supposed to fail without this fix? (because it doesn't fail when I try with the current master) (also the test name in the comment does not match)

Reply:
Yes, it's reported by pkg goleak that `found unexpected goroutines.`

If io error occurred, the leak happens. A closed segment will trigger an error.

Reply:
I apparently tested it the wrong way before. I was able to reproduce the error now by running the test on entire `wal` package instead of running a single test.